src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/CompareUtil.java

```diff
@@ -122,4 +125,41 @@ public char getOperator() {
             return '~';
         }
     }
+
+    /**
+     * Compares just like plain comparison, however, patterns are expanded using the
+     * given system environment settings.
+     */
+    class PlainVarCompareUtil implements CompareUtil {
+        private final EnvVars systemEnvVars;
+
+        public PlainVarCompareUtil() {
+            systemEnvVars = new EnvVars(EnvVars.masterEnvVars);
```

Comment:
Ah, I had missed the use of `EnvVars.masterEnvVars`. I don't think it's a good choice to put it here since you would then only pick up whatever is set when Jenkins starts up and not what is changed during runtime.
We need to find a way to provide it as close to the call to `matches` as possible, without having to create a new instance and copy over all values from `EnvVars.masterEnvVars` for every job check, since that could cause some performance issues.